### DISCUSSION TITLE
Module Service configuration settings (connection strings)

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#713] [General]
I'm implementing iServerStartup to get a RepoDB context established. Typically, I'd pull this from the appconfig.json during startup from the Iconfiguration parameter.  Where can I get access to the configuration in my module?

![image](https://user-images.githubusercontent.com/311063/91506983-712a1100-e888-11ea-9ee5-37fdddfa408c.png)

![image](https://user-images.githubusercontent.com/311063/91506937-58216000-e888-11ea-81c4-2490b4919664.png)

**Comment by sbwalker** [comment]
I am not sure why you are using IServerStartup? If you look at any of the sample modules or use the Module Creator you will see the pattern for connecting to a database from your module code.

**Reply by markdav-is** [reply]
I'm trying to access a different database and not using entity framework.  We'll be hooking into a few data sources and have about 6 connectionstrings total.  We are also using the integrated one for the modules and it works great!

**Comment by sbwalker** [comment]
Why not retrieve the connection string within your DBContext:

    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
      optionsBuilder.UseSqlServer(ConfigurationManager.ConnectionStrings["YourDatabase"].ConnectionString);
    }

**Reply by markdav-is** [reply]
I'll try doing that!

**Reply by markdav-is** [reply]
@sbwalker where are you getting your ConfigurationManager?  Is that from a DI constructor?

**Reply by sbwalker** [reply]
Yes you should always DI in .NET Core
